### PR TITLE
Feat/add configuration options

### PR DESCRIPTION
**Pull request by pixelass**
- allows setting patterns for environment creation
- allows deleting feature after merge
- allows setting the alias when merging
- allows setting version field & content-type
- allows setting migrations dir
- closes #21

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  INPUT_DELETE_FEATURE,
+  INPUT_SET_ALIAS,
+  INPUT_FEATURE_PATTERN,
+  INPUT_MASTER_PATTERN,
+  INPUT_VERSION_CONTENT_TYPE,
+  INPUT_VERSION_FIELD,
+} = process.env;
+
+export const DEFAULT_MIGRATIONS_DIR = "migrations";
+export const DEFAULT_MASTER_PATTERN = "master-[YYYY]-[MM]-[DD]-[mmss]";
```
Comment (pixelass) [inline src/constants.ts:17]:
```suggestion
export const DEFAULT_MASTER_PATTERN = "master-[YYYY]-[MM]-[DD]-[mm][ss]";
```

File: src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export const DEFAULT_FEATURE_PATTERN = "GH-[branch]";
+export const DEFAULT_VERSION_CONTENT_TYPE = "versionTracking";
+export const DEFAULT_VERSION_FIELD = "version";
+
```
Comment (pixelass) [inline src/constants.ts:21]:
```suggestion
export const INPUT_DELETE_FEATURE = false;
export const DEFAULT_SET_ALIAS = false;

```

File: src/constants.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+export const VERSION_CONTENT_TYPE =
+  INPUT_VERSION_CONTENT_TYPE || DEFAULT_VERSION_CONTENT_TYPE;
+export const FEATURE_PATTERN = INPUT_FEATURE_PATTERN || DEFAULT_FEATURE_PATTERN;
+export const MASTER_PATTERN = INPUT_MASTER_PATTERN || DEFAULT_MASTER_PATTERN;
+export const VERSION_FIELD = INPUT_VERSION_FIELD || DEFAULT_VERSION_FIELD;
```
Comment (pixelass) [inline src/constants.ts:26]:
```suggestion
export const VERSION_FIELD = INPUT_VERSION_FIELD || DEFAULT_VERSION_FIELD;
export const DELETE_FEATURE = INPUT_DELETE_FEATURE || DEFAULT_DELETE_FEATURE;
export const SET_ALIAS = INPUT_SET_ALIAS || DEFAULT_SET_ALIAS

```

File: src/action.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { runMigration } from "contentful-migration/built/bin/cli";
+import { readdir } from "fs";
+import path from "path";
+import { promisify } from "util";
+import {
+  CONTENTFUL_ALIAS,
+  INPUT_DELETE_FEATURE,
+  INPUT_SET_ALIAS,
```
Comment (pixelass) [inline src/action.ts:10]:
```suggestion
  DELETE_FEATURE,
  SET_ALIAS,
```

File: src/action.ts [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    Logger.success(
+      `Updated field ${VERSION_FIELD} in ${VERSION_CONTENT_TYPE} entry to ${migrationToRun}`
+    );
+  }
+
+  Logger.log(`Checking if we need to update ${CONTENTFUL_ALIAS} alias`);
+  // If the environmentId starts with ${CONTENTFUL_ALIAS} ("master")
+  // Then set the alias to the new environment
+  // Else inform the user
+  if (environmentId.startsWith(CONTENTFUL_ALIAS) && INPUT_SET_ALIAS) {
```
Comment (pixelass) [inline src/action.ts:170]:
```suggestion
  if (environmentId.startsWith(CONTENTFUL_ALIAS) && SET_ALIAS) {
```

File: src/action.ts [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    Logger.log("Running on feature branch");
+    Logger.log("No alias changes required");
+  }
+
+  // If the sandbox environment should be deleted
+  // And the baseRef is the repository default_branch (master|main ...)
+  // And the Pull Request has been merged
+  // Then delete the sandbox environment
+  if (
+    INPUT_DELETE_FEATURE &&
```
Comment (pixelass) [inline src/action.ts:191]:
```suggestion
    DELETE_FEATURE &&
```

File: src/action.ts [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+    branchNames.baseRef === branchNames.defaultBranch &&
+    github.context.payload.pull_request?.merged
+  ) {
+    try {
+      const environmentIdToDelete = `GH-${environmentNames.head}`;
```
Comment (pixelass) [inline src/action.ts:196]:
```suggestion
      const {head: environmentIdToDelete} = environmentNames;
```

File: README.md [hunk 8]
```diff
@@ -1,41 +1,116 @@
...
+* [Arguments](#arguments)
+* [Workflow](#workflow)
 
-See our [example usage](https://github.com/contentful-labs/contentful-action-example) in our contentful-action-example repository.
 
-This action requires a folder labeled migration inside your repo. You should place all your migrations in this directory.
+## Usage
 
-For this action to know which migrations it should run, we’ll need to track which migrations have been run by adding a version number into Contentful. We accomplish this in Contentful by creating a new content model with an ID of versionTracking that has a single short-text-field named version.
+This action runs migrations on your contentful space. PLease add your migration scripts to a directory called
```
Comment (dazulu) [inline README.md:23]:
```suggestion
This action runs migrations on your contentful space. Please add your migration scripts to a directory called
```

File: README.md [hunk 9]
```diff
@@ -1,41 +1,116 @@
...
-};
+module.exports = function () {};
 ```
 
-Going forward you can create a JavaScript file with an increasing integer such as `2.js`, `3.js` and so on. The action by default looks for a folder labeled `migration` but it's configurable via the environment variable `MIGRATIONS_DIR`.
+For every new version we can now increase the version (`2.js`, `3.js`, ...). Per default, this action looks for a
+directory labeled `migrations` but it's configurable via the arg `migrations_dir`.
 
-Lastly you'll need to update your workflow file to use this action and update the settings to include your `SPACE_ID` and `MANAGEMENT_API_KEY` from Contentful. Update your `main.yml` file by adding the following step:
+Next we can adjust our workflow file to use this action. YOu have to include your `space_id` and `management_api_key`
```
Comment (dazulu) [inline README.md:39]:
```suggestion
Next we can adjust our workflow file to use this action. You have to include your `space_id` and `management_api_key`
```

File: README.md [hunk 10]
```diff
@@ -1,41 +1,116 @@
...
+Please read the usage info above. The content-type and the field-id are configurable. 
+
+## Arguments
+
+Name | Type | Required | Default  | Description
+--- | --- | --- | --- | ---
+**space_id**             | `string`  | Yes | `undefined` | The id of the contentful space
+**management_api_key**   | `string`  | Yes | `undefined` | The management-api key for contentful
+delete_feature           | `boolean` | No  | `false` | Deletes sandbox environment if the head branch is merged
+set_alias                | `boolean` | No  | `false` | Aliases master the new master environment
```
Comment (dazulu) [inline README.md:89]:
```suggestion
set_alias                | `boolean` | No  | `false` | Aliases master to the new master environment
```

File: README.md [hunk 11]
```diff
@@ -1,41 +1,116 @@
...
+```yml
+- name: Contentful Migration
+  id: migrate
+  uses: contentful/contentful-migration-automation@1
+  with:
+    # delete_feature: true
+    # set_alias: true
+    # master_pattern: "main-[YY]-[MM]-[DD]-[hh]-[mm]"
+    # feature_pattern: "sandbox-[branch]"
+    # version_field: versionCounter
```
Comment (dazulu) [inline README.md:110]: versionCounter or versionTracking?

**Review by dazulu — APPROVED** [review]

**Pull request merged**